apps/speed.c

```diff
@@ -2953,6 +3050,105 @@ int speed_main(int argc, char **argv)
     }
 #endif                          /* OPENSSL_NO_RSA */
 
+#ifndef OPENSSL_NO_DH
+    for (testnum = 0; testnum < FFDH_NUM; testnum++) {
+        int ffdh_checks = 1;
+
+        if (!ffdh_doit[testnum])
+            continue;
+
+        for (i = 0; i < loopargs_len; i++) {
+            DH *key_A = NULL;
+            DH *key_B = NULL;
+            const BIGNUM *pubkey_A = NULL;
+            const BIGNUM *pubkey_B = NULL;
+            int secret_size;
+
+            /* Ensure that the error queue is empty */
+            if (ERR_peek_error()) {
+                BIO_printf(bio_err,
+                           "WARNING: the error queue contains previous unhandled errors.\n");
+                ERR_print_errors(bio_err);
+            }
+            key_A = DH_new_by_nid(ffdh_params[testnum].nid);
+            if (!key_A) {
+                BIO_printf(bio_err, "Error while initialising FFDH context.\n");
+                ERR_print_errors(bio_err);
+                rsa_count = 1;
+                ffdh_checks = 0;
+                break;
+            }
+            key_B = DH_new_by_nid(ffdh_params[testnum].nid);
+            if (!key_B) {
+                BIO_printf(bio_err, "Error while initialising FFDH context.\n");
+                ERR_print_errors(bio_err);
+                rsa_count = 1;
+                ffdh_checks = 0;
+                break;
+            }
+
+            if (1 != DH_generate_key(key_A) ||
+                1 != DH_generate_key(key_B)) {
```

Comment:
More usage of deprecated APIs here - so same question as above.

Reply:
fixed